src/connection_check.py

```diff
@@ -53,6 +53,35 @@ def broker_active(unit: Unit, zookeeper_config: Dict[str, str]) -> bool:
         True if broker id is recognised as active by ZooKeeper. Otherwise False.
     """
     broker_id = unit.name.split("/")[1]
+    brokers = brokers_active_get(zookeeper_config=zookeeper_config)
+    chroot = zookeeper_config.get("chroot", "")
+    result = broker_active_check(chroot=chroot, broker_id=broker_id, brokers=brokers)
+    return result
+
+
+def broker_active_check(chroot: str, broker_id: str, brokers: List[str]) -> bool:
+    """Checks ZooKeeper for client connections, checks for specific broker id.
+
+    Args:
+        chroot: chroot config from zookeeper
+        broker_id: the `Unit` to check connection of
+        brokers: the brokers relation data provided by ZooKeeper
+
+    Returns:
+        True if broker id is recognised as active by ZooKeeper. Otherwise False.
+    """
+    return f"{chroot}/brokers/ids/{broker_id}" in brokers
```

Comment:
For simplicity, I would suggest merging `broker_active_check()` into `broker_active()` for now. Every charm unit would be able to grab references to every other unit through the relation data, and it seems good to not restrict these checks to `broker_id`.  

```suggestion
    brokers = brokers_active_get(zookeeper_config=zookeeper_config)
    chroot = zookeeper_config.get("chroot", "")
    return f"{chroot}/brokers/ids/{broker_id}" in brokers
```

Reply:
Fixed.